clients/src/main/java/org/apache/kafka/common/record/OldLogEntry.java

```diff
@@ -0,0 +1,412 @@
+import static org.apache.kafka.common.record.Records.OFFSET_OFFSET;
+
+public abstract class OldLogEntry extends AbstractLogEntry implements LogRecord {
+
+    public abstract Record record();
+
+    @Override
+    public long lastOffset() {
+        return offset();
+    }
+
+    @Override
+    public boolean isValid() {
+        return record().isValid();
+    }
+
+    @Override
+    public void ensureValid() {
+        record().ensureValid();
+    }
+
+    @Override
+    public int keySize() {
+        return record().keySize();
+    }
+
+    @Override
+    public boolean hasKey() {
+        return record().hasKey();
+    }
+
+    @Override
+    public ByteBuffer key() {
+        return record().key();
+    }
+
+    @Override
+    public int valueSize() {
+        return record().valueSize();
+    }
+
+    @Override
+    public boolean hasNullValue() {
+        return record().hasNullValue();
+    }
+
+    @Override
+    public ByteBuffer value() {
+        return record().value();
+    }
+
+    @Override
+    public boolean hasMagic(byte magic) {
+        return magic == record().magic();
+    }
+
+    @Override
+    public boolean hasTimestampType(TimestampType timestampType) {
+        return record().timestampType() == timestampType;
+    }
+
+    @Override
+    public long checksum() {
+        return record().checksum();
+    }
+
+    @Override
+    public long timestamp() {
+        return record().timestamp();
+    }
+
+    @Override
+    public TimestampType timestampType() {
+        return record().timestampType();
+    }
+
+    @Override
+    public long baseOffset() {
+        return iterator().next().offset();
+    }
+
+    @Override
+    public byte magic() {
+        return record().magic();
+    }
+
+    @Override
+    public CompressionType compressionType() {
+        return record().compressionType();
+    }
+
+    @Override
+    public long sequence() {
+        return 0;
+    }
+
+    @Override
+    public int sizeInBytes() {
+        return record().sizeInBytes() + LOG_OVERHEAD;
+    }
+
+    @Override
+    public String toString() {
+        return "OldLogEntry(" + offset() + ", " + record() + ")";
+    }
+
+    @Override
+    public void writeTo(ByteBuffer buffer) {
+        writeHeader(buffer, offset(), record().sizeInBytes());
+        buffer.put(record().buffer().duplicate());
+    }
+
+    @Override
+    public long pid() {
+        return 0L;
+    }
+
+    @Override
+    public short epoch() {
+        return 0;
+    }
+
+    @Override
+    public int firstSequence() {
+        return 0;
+    }
+
+    @Override
+    public int lastSequence() {
+        return 0;
+    }
+
+    /**
+     * Get an iterator for the nested entries contained within this log entry. Note that
+     * if the entry is not compressed, then this method will return an iterator over the
+     * shallow entry only (i.e. this object).
+     * @return An iterator over the entries contained within this log entry
```

Comment:
Would it be more accurate to write `An iterator ovre the records contained within this log entry`?